lib/internal/test_runner/test.js

```diff
@@ -517,12 +524,10 @@ class Test extends AsyncResource {
         return;
       }
 
-      if (this.parent?.hooks.afterEach.length > 0) {
-        await this.parent[kRunHook]('afterEach', { args, ctx });
-      }
-
+      await afterEach();
       this.pass();
     } catch (err) {
+      await PromisePrototypeThen(PromiseResolve(afterEach()), noop, noop);
```

Comment:
Wouldn't it be simpler to use a try/catch in the `afterEach` function?
```suggestion
      await afterEach();
```
In any case, the `PromiseResolve` is not useful:
```suggestion
      await PromisePrototypeThen(afterEach(), noop, noop);
```